internal/controller/dnsrecord_controller.go

```diff
@@ -96,11 +97,25 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	}
 	dnsRecord := previous.DeepCopy()
 
+	managedZone := &v1alpha1.ManagedZone{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      dnsRecord.Spec.ManagedZoneRef.Name,
+			Namespace: dnsRecord.Namespace,
+		},
+	}
+	err = r.Get(ctx, client.ObjectKeyFromObject(managedZone), managedZone, &client.GetOptions{})
+	if err != nil {
```

Comment:
what if the managed zone is not found when the dns record is being deleted? The finalizer on the dnsrecord will never be removed

Reply:
Yes, this is the intention, that means that DNS Records have been created in the DNS Provider and have not been cleaned up.

Reply:
Well, you might be right. I know I do not know. But that is causing an issue anyway. Some tests are failing because the dns record is never gone after having deleted it.

Reply:
@philbrookes Is this the intention if a user deletes a `ManagedZone` first before deleting the `DNSRecords` ? 🤔 This is how the integration tests are in Kuadrant Operator. In the tests, the `ManagedZone` has completed their deletion, but `DNSRecords` remained, and prevented the test NS deletion. 

Seems like this introduces a required order of deletion for the user. So maybe the `ManagedZone` deletion should wait until the assoicated `DNSRecords` are deleted ? 🤔 

If this is the intended behaviour anyway, I've a fix to update the Kuadrant Operator `DNSPolicy` tests as part of a PR https://github.com/Kuadrant/kuadrant-operator/pull/689/commits/c70b5e1fc1e8372988a60aaa327452083a89b764. Just want to check this is the expected flow being merging

Reply:
Yes, this is the intention, if the DNS Record is deleted after the managed zone, the DNS Operator cannot remove the records from the DNS Provider, in order to highlight this to the user/admin the DNS Record never has it's finalizer removed.

See the original issue here: https://github.com/Kuadrant/dns-operator/issues/106